### PR TITLE
Docker file cleanup

### PR DESCRIPTION
**Pull request by ddingel**
## Description
- Moved to latest golang base image
- Removed go get as dependencies are already vendored
- Changed build to multistage to slim down resulting image from 412 to 11 MB
- Removed specialised ARM Dockerfile as golang and alpine are [multi arch Images](https://github.com/docker-library/official-images#architectures-other-than-amd64)

## Motivation and Context
- Lose weight after Christmas
- Should now support all officially supported platforms

## How Has This Been Tested?
- Build & Run on x86
- But I miss access to non x86 architecture

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by ddingel** [comment]
Maybe this change should go in first to solve the currently broken Travis Jobs?

**Comment by alexellis** [comment]
Have you tested this on ARM?

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-RUN go get -v -d github.com/docker/docker/api/types \
-    && go get -v -d github.com/docker/docker/api/types/filters \
-    && go get -v -d github.com/docker/docker/api/types/swarm \
-    && go get -v -d github.com/docker/docker/client \
-    && go get -v -d golang.org/x/net/context
 
 RUN mkdir -p /go/src/github.com/alexellis2/jaas
 WORKDIR /go/src/github.com/alexellis2/jaas
-COPY ./app.go ./
+COPY . .
```
Comment (alexellis) [inline Dockerfile]: Please don't copy `.` but the specific files / directories.

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
 
 RUN go build
 
-ENTRYPOINT ["./jaas"]
\ No newline at end of file
+FROM alpine
```
Comment (alexellis) [inline Dockerfile]: https://docs.docker.com/engine/userguide/eng-image/multistage-build/

> See my blog on multi-stage builds.. you should use a specific tag here such as "3.6"

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,14 @@
-FROM golang:1.7.3
+FROM golang:1.9.2-alpine3.7 AS build
 MAINTAINER alexellis2@gmail.com
-RUN go get -v -d github.com/docker/docker/api/types \
```
Comment (alexellis) [inline Dockerfile]: Why did you remove the `go get`?

**Comment by alexellis** [comment]
I've brought across the Dockerfile from OpenFaaS which results in a smaller binary. Please also see the contributing guide - this requires an issue to be raised/approved for each change / PR.
